jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/IoUtilTest.java

```diff
@@ -44,8 +47,46 @@ void findOpenPortWhenPortsAreBusy() throws IOException {
         assertThat(port2).isGreaterThan(port);
     }
 
+    @Test
+    void findOpenPortWithSmallAttemptsCount() throws IOException {
+        int port = IoUtil.getFreeRandomPort(30000, 60000, 30);
+        try (ServerSocket ss = new ServerSocket(port)) {
+            assertThat(ss).isNotNull();
+        }
+    }
+
+    @Test
+    void findOpenPortWithLargeAttemptsCount() throws IOException {
+        int port = IoUtil.getFreeRandomPort(30000, 60000, 1000);
+        try (ServerSocket ss = new ServerSocket(port)) {
+            assertThat(ss).isNotNull();
+        }
+    }
+
+    @Test
+    void invokeExceptionWhenCouldntFindPort() throws IOException {
+
+        // find open port to occupy
+        int foundPort = IoUtil.getFreeRandomPort(30000, 65000, 1000);
+
+        // use port
+        //noinspection resource
+        ServerSocket ignored = new ServerSocket(foundPort);
```

Comment:
This statement should be enclosed in a try-with-resources block so that the "ignored" port is closed when the test completes